mcp/src/main/java/io/modelcontextprotocol/server/McpAsyncServerExchange.java

```diff
@@ -126,7 +128,16 @@ public Mono<McpSchema.ElicitResult> createElicitation(McpSchema.ElicitRequest el
 	 * @return A Mono that emits the list of roots result.
 	 */
 	public Mono<McpSchema.ListRootsResult> listRoots() {
-		return this.listRoots(null);
+
+		return this.listRoots(McpSchema.FIRST_PAGE).expand(result -> {
```

Comment:
There are no tests for this? Can we have a test for 0 pages, 1 page, 5 pages? Network error after 3rd page?

Reply:
We do not have mcpclient that implements the root pagination.